sympy/physics/mechanics/abstract_body.py

```diff
@@ -0,0 +1,136 @@
+from abc import ABC, abstractmethod
```

Comment:
Named the file `abstract_body.py`, because we are currently using separate files for all bodies and this way we can avoid circular import errors. Another option would be to put the bodies in the same `bodies.py` file, but not sure if this is allowed with respect to backward compatibility. Besides of the fact that the file will be long.

P.S. A better name is also welcome.

Reply:
I think it's better to keep the files smaller as it makes navigating easier. So I support this being a separate file. I think I would prefer if there was consistency between the class name and the file name. If we're calling the ABC `_Body` then can this file be `_body.py`? Seeing as it shouldn't ever need to be used as a module by a user?

Reply:
Not sure if `_body.py` is the best name, because we also use the `_` prefix for experimental/sandbox modules.

Reply:
I agree with the point about the experimental/sandbox modules being prefixed with an underscore. I do think it's easier to navigate when classes and modules have the same name.

Reply:
Would you consider `body_abc.py` as better?

Reply:
Yes, I think I prefer `body_abc.py`. Would be nice if we followed a similar naming pattern for all ABCs. Or alternatively, we could aim to move `_Body` into `body.py` once `Body` has been deprecated and the circular dependency can be avoided another way. E.g. moving `Body` into `deprecated/body.py` or something.

Reply:
Not sure if a separate folder for deprecated modules is ideal. It does change the import path if someone were to be using `sympy.physics.mechanics.body` (not that I would encourage that).

When looking through SymPy for abstract base classes there does not seem to be a clear approach what we should do here. Also since almost none of them inherit `ABC`. Another option for naming would be `BodyBase`, which is in naming similar to `MatrixBase`, `StateBase`, `KetBase`, `BraBase`. An option would then also be to use `body_base.py`.

Reply:
> Not sure if a separate folder for deprecated modules is ideal. It does change the import path if someone were to be using sympy.physics.mechanics.body (not that I would encourage that).

Not sure that this is an issue if we're following the proper deprecation procedure. But this is now a discussion for a later date!

> When looking through SymPy for abstract base classes there does not seem to be a clear approach what we should do here. Also since almost none of them inherit `ABC`. Another option for naming would be `BodyBase`, which is in naming similar to `MatrixBase`, `StateBase`, `KetBase`, `BraBase`. An option would then also be to use `body_base.py`.

I think `BodyBase` and `body_base.py` would be my preference then if this is going to be the most consistent with elsewhere in SymPy.